app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  before_save { email.downcase! }
+
```

Comment:
group the code a little better. check the preferred way [here](https://github.com/rubocop-hq/rails-style-guide#activerecord).

In your case, put all your validations, then the callback and then the `has_secure_password`

Reply:
bookmarked it for further use. also, solved. ✓